CRM/Member/Form/Membership.php

```diff
@@ -1496,16 +1497,15 @@ public function submit() {
       $params['trxn_id'] = CRM_Utils_Array::value('trxn_id', $result);
       $params['is_test'] = ($this->_mode == 'live') ? 0 : 1;
       if (!empty($formValues['send_receipt'])) {
-        $params['receipt_date'] = $now;
+        $params['receipt_date'] = $params['receive_date'];
```

Comment:
@mattwire I went through the changes in this file (only) on top of https://github.com/civicrm/civicrm-core/pull/12642 which changes the same file & this line is the only one in this file I have a reservation about - I think it should be  date('YmdHis'); which is less confusing. 

I can pull the changes you have made to this file into #12642 if you want to review that & then I'll try to make sense of the changes in MembershipRenewal. The early ones seem pretty straightforward but the function extraction should probably stand alone

Reply:
@eileenmcnaughton I've reviewd #12642  When it is merged I'll update this PR